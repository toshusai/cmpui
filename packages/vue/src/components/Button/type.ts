export const ButtonVariant = {
  primary: "primary",
  secondary: "secondary",
} as const;

export type ButtonVariant = (typeof ButtonVariant)[keyof typeof ButtonVariant];

export const ButtonSize = {
  S: "S",
  M: "M",
  L: "L",
} as const;

export type ButtonSize = (typeof ButtonSize)[keyof typeof ButtonSize];

export type ButtonProps = {
  variant: ButtonVariant;
  size: ButtonSize;
  disabled: boolean;
};
