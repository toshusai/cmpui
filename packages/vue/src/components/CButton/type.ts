export const CButtonVariant = {
  primary: "primary",
  secondary: "secondary",
} as const;

export type CButtonVariant =
  (typeof CButtonVariant)[keyof typeof CButtonVariant];

export const CButtonSize = {
  S: "S",
  M: "M",
  L: "L",
} as const;

export type CButtonSize = (typeof CButtonSize)[keyof typeof CButtonSize];

export type CButtonProps = {
  variant?: CButtonVariant;
  size?: CButtonSize;
  disabled?: boolean;
};
