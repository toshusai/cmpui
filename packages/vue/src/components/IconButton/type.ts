export const IconButtonSize = {
  S: "S",
  M: "M",
} as const;

export type IconButtonSize =
  (typeof IconButtonSize)[keyof typeof IconButtonSize];

export type IconButtonProps = {
  bordered?: boolean;
  disabled?: boolean;
  rounded?: boolean;
  selected?: boolean;
  size?: IconButtonSize;
};
