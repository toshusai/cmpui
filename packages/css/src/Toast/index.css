.cmpui_toast__root {
  position: fixed;

  right: 16px;
  bottom: 16px;

  border: 1px solid var(--cmpui-primary-color);
  padding: 8px;
}

.cmpui_toast_transition-enter-from {
  transform: translateY(16px);
  opacity: 0;
}

.cmpui_toast_transition-enter-active {
  transition: opacity 175ms, transform 175ms;
}

.cmpui_toast_transition-enter-to {
  transform: translateY(0px);
  opacity: 1;
}

.cmpui_toast_transition-leave-from {
  transform: translateY(0px);
  opacity: 1;
}

.cmpui_toast_transition-leave-active {
  transition: opacity 175ms, transform 175ms;
}

.cmpui_toast_transition-leave-to {
  transform: translateY(16px);
  opacity: 0;
}
