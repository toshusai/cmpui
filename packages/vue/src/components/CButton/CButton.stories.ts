import type { Meta, StoryObj } from "@storybook/vue3";
import CButton from "./CButton.vue";
import { h } from "vue";

const meta = {
  title: "inputs/CButton",
  tags: ["autodocs"],
  component: CButton,
} satisfies Meta<typeof CButton>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    default: "Button",
    variant: "primary",
  },
};

export const Variants: Story = {
  render: () =>
    h(() => [
      h(CButton, { variant: "primary" }, () => "Primary"),
      h(CButton, { variant: "secondary" }, () => "Secondary"),
      h(CButton, { variant: "transparent" }, () => "Transparent"),
    ]),
};

export const Sizes: Story = {
  render: () =>
    h(() => [
      h(CButton, { size: "M" }, () => "Medium"),
      h(CButton, { size: "S" }, () => "Small"),
    ]),
};

export const Disabled: Story = {
  render: () =>
    h(() => [
      h(CButton, { variant: "primary", disabled: true }, () => "Primary"),
      h(CButton, { variant: "secondary", disabled: true }, () => "Secondary"),
    ]),
};

export const Bordered: Story = {
  render: () =>
    h(() => [
      h(CButton, { variant: "primary", bordered: true }, () => "Primary"),
      h(CButton, { variant: "secondary", bordered: true }, () => "Secondary"),
    ]),
};
