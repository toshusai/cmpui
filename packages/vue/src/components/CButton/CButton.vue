<script setup lang="ts">
import { CButtonProps } from "./type";

withDefaults(defineProps<CButtonProps>(), {
  variant: "primary",
  size: "M",
  disabled: false,
});
</script>

<template>
  <button
    class="cmpui_button"
    :data-variant="variant"
    :data-size="size"
    :disabled="disabled"
    v-bind="$attrs"
  >
    <slot></slot>
  </button>
</template>
