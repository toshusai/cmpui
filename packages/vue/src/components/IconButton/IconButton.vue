<script setup lang="ts">
import { IconButtonProps } from "./type";

withDefaults(defineProps<IconButtonProps>(), {
  size: "M",
});
</script>

<template>
  <button
    class="cmpui_icon-button__root"
    :aria-selected="selected"
    :data-size="size"
    :disabled="disabled"
    :data-rounded="rounded"
    :data-border="border"
  >
    <slot></slot>
  </button>
</template>
