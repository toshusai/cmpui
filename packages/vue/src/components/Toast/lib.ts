import { Static, Transition, VNode } from "vue";

export function isValidVNode(node: VNode): node is VNode {
  switch (node.type) {
    case Comment:
      return false;
    case Static:
      return false;
    case Transition:
      return false;
    default:
      return true;
  }
}

export const TOAST_PROVIDER_KEY = Symbol("toast-provider");
