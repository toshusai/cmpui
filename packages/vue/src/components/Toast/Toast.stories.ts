import type { Meta, StoryObj } from "@storybook/vue3";
import IconButton from "../IconButton/IconButton.vue";
import CButton from "../CButton/CButton.vue";
import ToastProvider from "./ToastProvider.vue";
import Toast from "./Toast.vue";

const meta = {
  title: "components/Toast",
  component: Toast,
  args: {
    text: "tooltip",
  },
} satisfies Meta<typeof Toast>;

export default meta;
type Story = StoryObj<typeof meta>;

const html = String.raw;

export const Default: Story = {
  render: () => ({
    components: { CButton, Toast, ToastProvider },
    template: `
        <Toast text="Tooltip">
          <CButton>
            Hover me
          </CButton>
        </Toast>
    `,
  }),
};

export const Flip: Story = {
  render: () => ({
    components: { CButton, Toast, ToastProvider },
    template: `
        <Toast text="Tooltip">
          <CButton style="position: absolute; bottom: 4px; transform: translateX(-50%);">
            Hover me
          </CButton>
        </Toast>
    `,
  }),
};

export const AutoPlacement: Story = {
  render: () => ({
    components: { Button: CButton, Toast, ToastProvider },
    template: `
        <Toast text="Tooltip can be placed anywhere">
          <CButton style="position: absolute; bottom: 4px; right: 4px">
            Hover me
          </CButton>
        </Toast>
    `,
  }),
};

export const MultipleItems: Story = {
  render: () => ({
    components: { IconButton, Toast, ToastProvider },
    template: html`
      <ToastProvider>
        <div class="cmpui_icon-button__group">
          <Toast text="pointer">
            <IconButton>
              <i class="ti ti-pointer"></i>
            </IconButton>
          </Toast>
          <Toast text="typography">
            <IconButton>
              <i class="ti ti-typography"></i>
            </IconButton>
          </Toast>
          <Toast text="italic">
            <IconButton>
              <i class="ti ti-italic"></i>
            </IconButton>
          </Toast>
        </div>
      </ToastProvider>
    `,
  }),
};
