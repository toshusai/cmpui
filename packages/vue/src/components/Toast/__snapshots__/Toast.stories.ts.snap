// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Toast AutoPlacement smoke-test 1`] = `
<cbutton aria-describedby="id-3"
         style="position: absolute; bottom: 4px; right: 4px;"
>
  Hover me
</cbutton>
`;

exports[`components/Toast Default smoke-test 1`] = `
<button class="cmpui_button"
        data-variant="primary"
        data-size="M"
        aria-describedby="id-1"
>
  Hover me
</button>
`;

exports[`components/Toast Flip smoke-test 1`] = `
<button class="cmpui_button"
        data-variant="primary"
        data-size="M"
        aria-describedby="id-2"
        style="position: absolute; bottom: 4px; transform: translateX(-50%);"
>
  Hover me
</button>
`;

exports[`components/Toast MultipleItems smoke-test 1`] = `
<div class="cmpui_icon-button__group">
  <button class="cmpui_icon-button__root"
          aria-selected="false"
          data-size="M"
          data-rounded="false"
          data-border="false"
          aria-describedby="id-4"
  >
    <i class="ti ti-pointer">
    </i>
  </button>
  <button class="cmpui_icon-button__root"
          aria-selected="false"
          data-size="M"
          data-rounded="false"
          data-border="false"
          aria-describedby="id-5"
  >
    <i class="ti ti-typography">
    </i>
  </button>
  <button class="cmpui_icon-button__root"
          aria-selected="false"
          data-size="M"
          data-rounded="false"
          data-border="false"
          aria-describedby="id-6"
  >
    <i class="ti ti-italic">
    </i>
  </button>
</div>
`;
